### ISSUE TITLE
For a Thin, the os/2 value for static TTFs can be 100 or 250

### ISSUE THREAD

**Issue by felipesanches** [googlefonts/fontbakery#2644]
I received this request from @davelab6 by email regarding an issue related to a FB report for the Big Shoulders family, by @yanone
> +Felipe can you file a bug on FontBakery issue tracker that for a Thin, the os/2 value for static TTFs can be 100 (to match the wght axis value of a VF) or 250, but for cffs, it should be 250. 

@m4rc1e also said:
> Fontbakery currently produces a WARN if a Thin weight has a weightClass value of 100, https://github.com/googlefonts/fontbakery/blob/master/Lib/fontbakery/profiles/googlefonts.py#L834-L845.
> 
> As Dave said, there is nothing wrong with a ttf Thin weight having a weightClass of 100. I set it to 100 since our VF Thin instances uses 100. By having both the static fonts and VF instance using 100, it means I wouldn't have to do any post processing fixes once the fonts had been generated.

to which @davelab6 replied:
> Exactly. FB needs urgently fixing to prevent all false negatives.

I will inspect this and provide a fix asap.

**Comment by felipesanches** [comment]
`com.google.fonts/check/usweightclass` does have already this kind of logic:

![Screenshot at 2019-09-26 18:25:44](https://user-images.githubusercontent.com/213676/65726508-c53edb00-e0a4-11e9-8e18-351ff704aedb.png)

`expected_style` is the output of @m4rc1e 's `style_parser` class.

The reason why the "thin can be 100 ou 250" message was not displayed was because of the "10pt" in the stylename. I am not yet sure if this is a shortcoming of Marc's style parser or something else.

**Comment by felipesanches** [comment]
@m4rc1e, how should we proceed here?

**Comment by m4rc1e** [comment]
> The reason why the "thin can be 100 or 250" message was not displayed was because of the "10pt" in the stylename.

This is correct. Since the font we're checking is a VF with two axes, the style name (nameId 17) contains both the `opsz` and `wght` values (10pt Thin). Unfortunately, this approach is not compatible with the GF static api which only accepts `wght` values in the style name (the current implementation).

Let me think about this issue for a while. I don't really like the current implementation because we're checking the style name in this check when imo we shouldn't.

**Comment by tphinney** [comment]
This same issue also affects ExtraLight in OT-CFF, for the same reason. It is 200 in TTF, but needs to be 250+ in OT-CFF. (If there is no Thin, it could be 250, otherwise 275 is reasonable. Could just require 275 all the time.)

**Comment by RosaWagner** [comment]
The warning is a bit confusing, this what I understand:

When exporting static TTF and CFF at the same time: I have to set up the custom parameters to 250(Thin) and 275(Xlight) _even if the specs require 100(Thin) and 200(XLight) for ttf._
Then, if I have to export a variable font with the same build, Fontmake will map it correctly taking the axis coordinate and the OS/2 us_WeightClass value.

Did I understand it correctly?

**Comment by vv-monsalve** [comment]
@RosaWagner is your question still in place?

But, above all, is the 250 for OT-CFF Thin requirement still in place at all? In the GF-Guide we are saying:

> Note for OTF-CFF fonts:
>
> Some fonts with a CSS font-weight property of 100 or 200 then the usWeightClass value may be 250, as values of 249 or less in CFF fonts suffered from blurry rendering on older versions of Windows — but in 2020 this is no longer necessary.

**Comment by RosaWagner** [comment]
I think my question was for gftools or fontmake, but I was just starting in June 2020. 

Lots of users still use previous versions of Windows, so the check is a good thing imo. Although it could be updated this way:

if font is OTF-CFF
if Thin == 250 and ExtraLight == 275
PASS

if Thin == 100  and ExtraLight == 200 
WARN for backward compatibility with Windows (specify here which version), consider having Thin == 250 and ExtraLight == 275 to avoid blablabla

if Thin =! 100/250 and ExtraLight =! 200/275
FAIL cause not the correct usWeightClass
